src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java

```diff
@@ -0,0 +1,90 @@
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction;
+import com.intellij.codeInsight.intention.preview.IntentionPreviewInfo;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.psi.*;
+import com.intellij.util.IncorrectOperationException;
+import com.weirddev.testme.intellij.action.CreateTestMeAction;
+import com.weirddev.testme.intellij.action.TestMeActionHandler;
+import com.weirddev.testme.intellij.builder.MethodFactory;
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * create test for selected method
+ *
+ * @author huangliang
+ */
+@NonNls
+final class CreateTestForMethodIntention extends PsiElementBaseIntentionAction implements IntentionAction {
+
+    /**
+     * Checks whether this intention is available at the caret offset in file - the caret must sit a testable method
+     * character in a ternary statement. If this condition is met, this intention's entry is shown in the available
+     * intentions list.
+     *
+     * <p>Note: this method must do its checks quickly and return.</p>
+     *
+     * @param project a reference to the Project object being edited.
+     * @param editor  a reference to the object editing the project source
+     * @param element a reference to the PSI element currently under the caret
+     * @return {@code true} if the caret is in a literal string element, so this functionality should be added to the
+     * intention menu or {@code false} for all other types of caret positions
+     */
+    public boolean isAvailable(@NotNull Project project, Editor editor, @Nullable PsiElement element) {
+        // Quick sanity check
+        return null != element && element instanceof PsiIdentifier && element.getParent() instanceof PsiMethod;
+    }
+
+    /**
+     * pop up the testMe menu list
+     * @param project a reference to the Project object being edited.
+     * @param editor  a reference to the object editing the project source
+     * @param element a reference to the PSI element currently under the caret
+     * @throws IncorrectOperationException Thrown by underlying (PSI model) write action context
+     *                                     when manipulation of the PSI tree fails.
+     */
+    public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element)
+        throws IncorrectOperationException {
+        PsiMethod method = (PsiMethod)element.getParent();
+        final PsiClass srcClass = CreateTestMeAction.getContainingClass(element);
+        // if is not testable show a error dialog
+        if (!MethodFactory.isTestable(method, srcClass)) {
+            String message = "Method \"" + method.getName() + "\" is not testable!";
+            String title = "TestMe: Method Not Testable";
+            Messages.showErrorDialog(message, title);
```

Comment:
I like the idea of popping an error notification in such case 👍 
but currently it results in an error:
```
java.lang.Throwable: AWT events are not allowed inside write action: java.awt.event.FocusEvent[FOCUS_LOST,temporary,opposite=EditorComponent
at com.intellij.openapi.ui.Messages.showErrorDialog(Messages.java:636)  
at com.weirddev.testme.intellij.intention.CreateTestForMethodIntention.invoke(CreateTestForMethodIntention.java:60)
```
instead use error hint, UX also seems a bit more friendly. see JB implementation for a very similar use case - https://github.com/JetBrains/intellij-community/blob/e0370874cac76407163f9f5ad68f479ac03c173f/java/java-impl/src/com/intellij/testIntegration/createTest/GenerateMissedTestsAction.java#L67

furthermore, I think it would be good to re use some fast checks as in `GenerateMissedTestsAction.isAvailable`  from file referenced above . like verify method is public and not abstract